ddsrouter_core/src/cpp/types/participant/ParticipantKind.cpp

```diff
@@ -30,114 +30,43 @@ namespace ddsrouter {
 namespace core {
 namespace types {
 
-const std::map<ParticipantKindType, std::vector<std::string>> ParticipantKind::participant_kind_with_aliases_ =
-{
-    {PARTICIPANT_KIND_INVALID, {"__invalid_participant_kind__"}},
-    {VOID, {"void"}},
-    {ECHO, {"echo"}},
-    {DUMMY, {"dummy"}},
-    {SIMPLE_RTPS, {"local", "simple"}},
-    {LOCAL_DISCOVERY_SERVER, {"discovery-server", "ds", "local-ds", "local-discovery-server"}},
-    {WAN, {"wan", "router"}},
-};
-
-ParticipantKind::ParticipantKind(
-        ParticipantKindType value) noexcept
-    : value_(value)
-{
-}
-
-ParticipantKind::ParticipantKind() noexcept
-    : ParticipantKind(PARTICIPANT_KIND_INVALID)
-{
-}
-
-bool ParticipantKind::is_valid() const noexcept
-{
-    return value_ != PARTICIPANT_KIND_INVALID;
-}
-
-std::string ParticipantKind::to_string() const noexcept
-{
-    auto it = ParticipantKind::participant_kind_with_aliases_.find(value_);
-
-    // Value must be in map
-    assert(it != ParticipantKind::participant_kind_with_aliases_.end());
-
-    return it->second[0];
-}
-
-ParticipantKindType ParticipantKind::operator ()() const noexcept
-{
-    return value_;
-}
-
-bool ParticipantKind::operator <(
-        const ParticipantKind& other) const noexcept
-{
-    return this->value_ < other.value_;
-}
-
-bool ParticipantKind::operator ==(
-        const ParticipantKind& other) const noexcept
+std::ostream& operator <<(
+        std::ostream& os,
+        ParticipantKind kind)
 {
-    return (*this)() == other();
-}
+    try
+    {
 
-ParticipantKind ParticipantKind::participant_kind_from_name(
-        std::string kind) noexcept
-{
-    // Pass kind name to lowercase
-    utils::to_lowercase(kind);
+        os << ParticipantKindStrings.at(static_cast<ParticipantKindType>(kind));
 
-    // Look for string in names map and return the kind it matches
-    for (auto kinds : ParticipantKind::participant_kind_with_aliases_)
+    }
+    catch (const std::out_of_range& oor)
     {
-        // Check if it fix with any of the aliases
-        for (std::string alias : kinds.second)
-        {
-            if (kind == alias)
-            {
-                return kinds.first;
-            }
-        }
+        utils::tsnh(utils::Formatter() << "Invalid Participant Kind." << static_cast<ParticipantKindType>(kind));
     }
-    return PARTICIPANT_KIND_INVALID;
+    return os;
 }
 
-std::vector<ParticipantKind> ParticipantKind::all_valid_participant_kinds() noexcept
+ParticipantKind participant_kind_from_name(
+        std::string participantKindName)
 {
-    std::vector<ParticipantKind> result;
 
-    for (auto kind : participant_kind_with_aliases_)
+    if (participantKindName.size() == 0)
```

Comment:
Add comment explaining why this. Is it because it would make function faster in this case, or because otherwise it will break?

Reply:
Added comments. It just does the same as previous function, but it is both faster and safer. Overall, previous approach involved a vector creation, while new proposal just loops over a compile-time array with bounds checking and additional consistency tests for compile-time array sizes.